File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/MainActivity.java
package de.szalkowski.activitylauncher;

import android.app.SearchManager;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.database.Cursor;
import android.database.MatrixCursor;
import android.os.AsyncTask;
import android.os.Bundle;
import android.provider.BaseColumns;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Filter;
import android.widget.Filterable;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.cursoradapter.widget.SimpleCursorAdapter;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.preference.PreferenceManager;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    private static final int SEARCH_QUERY_THRESHOLD = 2;

    private SharedPreferences prefs;
    private String localeString;
    private SearchView searchView;
    private Filterable filterTarget = null;
    private String filter = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle(R.string.app_name);

        this.prefs = PreferenceManager.getDefaultSharedPreferences(getBaseContext());

        this.localeString = prefs.getString("language", "System Default");
        Configuration config = SettingsUtils.createLocaleConfiguration(this.localeString);
        getBaseContext().getResources().updateConfiguration(config,
                getBaseContext().getResources().getDisplayMetrics());

        if (!this.prefs.getBoolean("disclaimer_accepted", false)) {
            DialogFragment dialog = new DisclaimerDialogFragment();
            dialog.show(getSupportFragmentManager(), "DisclaimerDialogFragment");
        }

        AllTasksListFragment fragment = new AllTasksListFragment();
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, fragment).commit();
        this.filterTarget = fragment;
    }

    @Override
    public void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        // Restore the previously serialized current dropdown position.
        super.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);

        searchView = (SearchView) menu.findItem(R.id.search).getActionView();
        searchView.setQueryHint(this.getText(R.string.filter_hint));
        searchView.setSuggestionsAdapter(new SimpleCursorAdapter(
                MainActivity.this, android.R.layout.simple_list_item_1, null,
                new String[] { SearchManager.SUGGEST_COLUMN_TEXT_1 },
                new int[] { android.R.id.text1 }));

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                if (query.length() >= SEARCH_QUERY_THRESHOLD) {
                    new AddSearchSuggestionTask().execute(query);
                }
                return true;
            }

            @Override
            public boolean onQueryTextChange(String query) {
                if (query.length() >= SEARCH_QUERY_THRESHOLD) {
                    new FetchSearchSuggestionsTask().execute(query);
                } else {
                    searchView.getSuggestionsAdapter().changeCursor(null);
                }

                MainActivity.this.filter = query;
                MainActivity.this.updateFilter(query);
                return true;
            }
        });

        searchView.setOnSuggestionListener(new SearchView.OnSuggestionListener() {

            @Override
            public boolean onSuggestionSelect(int position) {
                Cursor cursor = (Cursor) searchView.getSuggestionsAdapter().getItem(position);
                CharSequence value = cursor.getString(cursor.getColumnIndexOrThrow(SearchManager.SUGGEST_COLUMN_TEXT_1));
                searchView.setQuery(value, false);

                return true;
            }

            @Override
            public boolean onSuggestionClick(int position) {
                return onSuggestionSelect(position);
            }
        });

        return true;
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (!this.localeString.equals(this.prefs.getString("language", "System Default"))) {
            recreateFragments();
        }
        updateFilter(this.filter);
    }

    private void recreateFragments() {
        recreate();
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            getSupportFragmentManager().beginTransaction().remove(fragment).commit();
        }
    }

    private void updateFilter(String query) {
        Filter filter = this.filterTarget.getFilter();
        if (filter != null) {
            filter.filter(query);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            Intent intent = new Intent(this, SettingsActivity.class);
            startActivity(intent);
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        // Serialize the current dropdown position.
        super.onSaveInstanceState(outState);
    }

    public boolean isRootAllowed() {
        return PreferenceManager.getDefaultSharedPreferences(getBaseContext()).getBoolean("allow_root", false);
    }

    // Добавить новую строку в историю
    private class AddSearchSuggestionTask extends AsyncTask<String, Void, Void> {

        @Override
        protected Void doInBackground(String... params) {
            Db.insertItem(MainActivity.this, params[0]);
            return null;
        }
    }

    // Достать историю поиска по введенной строке
    private class FetchSearchSuggestionsTask extends AsyncTask<String, Void, Cursor> {

        private final String[] sAutocompleteColNames = new String[] {
                BaseColumns._ID,
                SearchManager.SUGGEST_COLUMN_TEXT_1
        };

        @Override
        protected Cursor doInBackground(String... params) {
            MatrixCursor cursor = new MatrixCursor(sAutocompleteColNames);

            List<String> queries = Db.getItems(MainActivity.this, params[0]);
            for (int index = 0; index < queries.size(); index++) {
                Object[] row = new Object[] { index, queries.get(index) };
                cursor.addRow(row);
            }

            return cursor;
        }

        @Override
        protected void onPostExecute(Cursor result) {
            searchView.getSuggestionsAdapter().changeCursor(result);
        }
    }
}

File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/MainActivity.kt
package de.szalkowski.activitylauncher

import android.annotation.TargetApi
import android.app.ActionBar
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem

import androidx.fragment.app.DialogFragment
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

class MainActivity : FragmentActivity() {

    protected val LOG = "de.szalkowski.activitylauncher.MainActivity"

    /**
     * Backward-compatible version of [ActionBar.getThemedContext] that
     * simply returns the [android.app.Activity] if
     * `getThemedContext` is unavailable.
     */
    private val actionBarThemedContextCompat: Context
        @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
        get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            actionBar!!.themedContext
        } else {
            this
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (!getPreferences(Context.MODE_PRIVATE).getBoolean("disclaimer_accepted", false)) {
            val dialog = DisclaimerDialogFragment()
            dialog.show(supportFragmentManager, "DisclaimerDialogFragment")
        }

        val fragment = AllTasksListFragment()
        supportFragmentManager.beginTransaction()
                .replace(R.id.container, fragment).commit()
    }

    public override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        // Restore the previously serialized current dropdown position.
        super.onRestoreInstanceState(savedInstanceState)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_view_source -> {
                val i2 = Intent(Intent.ACTION_VIEW)
                i2.data = Uri.parse(this.getString(R.string.url_source))
                this.startActivity(i2)
                return true
            }

            R.id.action_view_translation -> {
                val i3 = Intent(Intent.ACTION_VIEW)
                i3.data = Uri.parse(this.getString(R.string.url_translation))
                this.startActivity(i3)
                return true
            }

            R.id.action_view_bugs -> {
                val i4 = Intent(Intent.ACTION_VIEW)
                i4.data = Uri.parse(this.getString(R.string.url_bugs))
                this.startActivity(i4)
                return true
            }

            else -> return super.onOptionsItemSelected(item)
        }
    }


    public override fun onSaveInstanceState(outState: Bundle) {
        // Serialize the current dropdown position.
        super.onSaveInstanceState(outState)
    }
}
